liquidctl/driver/krakenz3.py

```diff
@@ -0,0 +1,495 @@
+            (_STATUS_PUMP_DUTY, msg[19], "%"),
+            (_STATUS_FAN_SPEED, msg[24] << 8 | msg[23], "rpm"),
+            (_STATUS_FAN_DUTY, msg[25], "%"),
+        ]
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to use a speed duty profile."""
+
+        cid, dmin, dmax = self._speed_channels[channel]
+        header = [0x72, cid, 0x00, 0x00]
+        norm = normalize_profile(profile, _CRITICAL_TEMPERATURE)
+        stdtemps = list(range(20, _CRITICAL_TEMPERATURE + 1))
+        interp = [clamp(interpolate_profile(norm, t), dmin, dmax) for t in stdtemps]
+        for temp, duty in zip(stdtemps, interp):
+            _LOGGER.info(
+                "setting %s PWM duty to %d%% for liquid temperature >= %d°C", channel, duty, temp
+            )
+        self._write(header + interp)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set channel to a fixed speed duty."""
+
+        self.set_speed_profile(channel, [(0, duty), (_CRITICAL_TEMPERATURE - 1, duty)])
+
+    def set_color(self, channel, mode, color, speed="normal", direction="forward", **kwargs):
+        """
+        supported channel is screen,
+        mode is an array ["config|static|liquid", "path to image|brightness|orientation", "brightness,orientation value"]
+        color is unused,
+        speed is unused,
+        direction is unused
+        """
+        assert channel == "screen", f"Invalid Channel: {channel}"
+
+        # get orientation and brightness
+        if not self.initialized:
+            self._write([0x30, 0x01])
+
+            def parse_lcd_info(msg):
+                self.brightness = msg[0x18]
+                on = msg[0x1A]  # orientation number
+                self.orientation = on
+
+            self._read_until({b"\x31\x01": parse_lcd_info})
+
+        if len(mode) > 2:
+            if mode[0].lower() == "config" and len(mode) > 2:
+                if mode[1].lower() == "brightness":
+                    assert int(mode[2]) >= 0 and int(mode[2]) <= 100, "Invalid brightness value"
+                    self._write([0x30, 0x02, 0x01, int(mode[2]), 0x0, 0x0, 0x1, self.orientation])
+                    return
+                elif mode[1].lower() == "orientation":
+                    assert int(mode[2]) >= 0 and int(mode[2]) <= 3, "Invalid orientation value"
+                    self._write([0x30, 0x02, 0x01, self.brightness, 0x0, 0x0, 0x1, int(mode[2])])
+                    return
+        elif len(mode) > 1:
+            if mode[0].lower() == "static":
+                data = self._prepare_static_file(mode[1], self.orientation)
+                self.send_data(data, [0x02, 0x0, 0x0, 0x0, 0x0, 0x40, 0x06])
+                return
+            elif mode[0].lower() == "gif":
+                # send loading animation first
+                loading_animation = open(os.path.join(pathlib.Path(__file__).parent.resolve(), 'loading.gif'), 'rb')
```

Comment:
it was a gif file in the same driver directory. the CAM software displays a similar loading gif before it transmits the user gif so I wanted to keep the same experience but I don't think its needed since it just adds time sending the loading gif. removed